logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationsAndConnectionsUseCaseTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.feature.connection.ObserveConnectionListUseCase
+import com.wire.kalium.logic.framework.TestConversationDetails
+import com.wire.kalium.logic.sync.SyncManager
+import io.mockative.Mock
+import io.mockative.configure
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+class ObserveConversationsAndConnectionsUseCaseTest {
+
+    @Mock
+    val observeConversationListDetailsUseCase: ObserveConversationListDetailsUseCase =
+        mock(ObserveConversationListDetailsUseCase::class)
+
+    @Mock
+    val observeConnectionListUseCase: ObserveConnectionListUseCase = mock(ObserveConnectionListUseCase::class)
+
+    @Mock
+    val syncManager: SyncManager = configure(mock(SyncManager::class)) { stubsUnitByDefault = true }
+
+    private lateinit var observeConversationsAndConnectionsUseCase: ObserveConversationsAndConnectionsUseCase
+
+    @BeforeTest
+    fun setup() {
+        observeConversationsAndConnectionsUseCase =
+            ObserveConversationsAndConnectionsUseCaseImpl(syncManager, observeConversationListDetailsUseCase, observeConnectionListUseCase)
+    }
+
+    @Test
+    fun givenSomeConversationsAndConnections_whenObservingDetailsListAndConnections_thenObserveConversationListShouldBeCalled() = runTest {
```

Comment:
Even tough, currently we are not testing error cases on flow's use cases, because we just bubble up the exception, I added a test case for that. 

Also, I didn't add the arrangement helper class for this, since the class was small to do it here and no duplication was made